library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -1215,15 +1215,16 @@ impl<T, A: Allocator> VecDeque<T, A> {
         unsafe { IterMut::new(ring, tail, head, PhantomData) }
     }
 
-    /// Creates a draining iterator that removes the specified range in the
-    /// `VecDeque` and yields the removed items.
+    /// Removes the specified range from the `VecDeque`, returning all removed
+    /// elements as an iterator.
```

Comment:
I kind-of prefer the original sentence here. What’s the motivation for the change? The call to `drain` itself does not do anything (yet) beyond creating an iterator. The returned iterator has a lifetime coupled to the mutable borrow; for an operation that “Removes the specified range … returning all removed elements as an iterator” one could expect an iterator that owns the elements and can be created with a shorter mutable borrow not coupled to a lifetime parameter of the type of the iterator.

The same applies to `Vec` and `String`

Reply:
This is in the commit for #92902 and should be discussed there or in #92765.

Reply:
Oh, I see. I didn't realize that there's another unmerged PR.

Reply:
And I didn't realize anyone would notice this PR this month.

Reply:
I just happened to look at the list of latest/new open PRs today, (mainly because I have an open PR myself where I'm a bit impatiently checking whether anything has happened w.r.t. review yet,) then happend to be quickly looking into this one for whatever reason and probably wouldn't even have started any review if I hadn't spotted the typo of "closure". 😅 Please don't mind my lengthy side-discussion about validity of collection types that I opened below 🙈